--- C:/msys/1.0/home/Administrator/portaudio-snapshot-clean/src/hostapi/dsound/pa_win_ds.c	Wed Jan 21 07:07:32 2009
+++ C:/msys/1.0/home/Administrator/portaudio/src/hostapi/dsound/pa_win_ds.c	Sat Feb 21 23:36:13 2009
@@ -95,8 +95,9 @@
 #include "pa_win_ds.h"
 #include "pa_win_ds_dynlink.h"
 #include "pa_win_waveformat.h"
+#ifdef PAWIN_USE_WDMKS_DEVICE_INFO
 #include "pa_win_wdmks_utils.h"
-
+#endif
 
 #if (defined(WIN32) && (defined(_MSC_VER) && (_MSC_VER >= 1200))) /* MSC version 6 and above */
 #pragma comment( lib, "dsound.lib" )
--- C:/msys/1.0/home/Administrator/portaudio-snapshot-clean/configure	Thu Mar 06 12:14:06 2008
+++ C:/msys/1.0/home/Administrator/portaudio/configure	Sat Feb 21 23:51:24 2009
@@ -21410,10 +21410,10 @@
             PADLL="portaudio.dll"
             THREAD_CFLAGS="-mthreads"
             SHARED_FLAGS="-shared"
-            DLL_LIBS="${DLL_LIBS} -lwinmm -lm -L./dx7sdk/lib -ldsound -lole32"
+            DLL_LIBS="${DLL_LIBS} -lwinmm -lm -L$DXDIR/lib -ldsound -lole32"
             #VC98="\"/c/Program Files/Microsoft Visual Studio/VC98/Include\""
             #CFLAGS="$CFLAGS -I$VC98 -DPA_NO_WMME -DPA_NO_ASIO"
-            CFLAGS="$CFLAGS -I\$(top_srcdir)/include -I$DXDIR/include -DPA_NO_WMME -DPA_NO_ASIO" -DPA_NO_WDMKS
+            CFLAGS="$CFLAGS -I\$(top_srcdir)/include -I$DXDIR/include -DPA_NO_WMME -DPA_NO_ASIO -DPA_NO_WDMKS"
         elif [ $with_winapi = "asio" ] ; then
             if [ $with_asiodir ] ; then
               ASIODIR="$with_asiodir"
