--- MultiUser.nsh.orig  2017-09-16 20:03:48.000000000 +0000
+++ MultiUser.nsh       2017-09-16 20:14:24.236880542 +0000
@@ -133,7 +133,11 @@
         ${if} ${AtLeastWin2000}
           StrCpy $INSTDIR "$LOCALAPPDATA\${MULTIUSER_INSTALLMODE_INSTDIR}"
         ${else}
-          StrCpy $INSTDIR "$PROGRAMFILES\${MULTIUSER_INSTALLMODE_INSTDIR}"
+         !ifdef MULTIUSER_USE_PROGRAMFILES64
+            StrCpy $INSTDIR "$PROGRAMFILES64\${MULTIUSER_INSTALLMODE_INSTDIR}"
+          !else
+            StrCpy $INSTDIR "$PROGRAMFILES\${MULTIUSER_INSTALLMODE_INSTDIR}"
+         !endif
         ${endif}
       !endif
     !endif
